### PR TITLE
[PyROOT][DF] Improve RDataFrame.AsNumpy performance

### PR DESCRIPTION
**Pull request by stwunsch**
* Fixes a warning in newer python version (first commit)
* Adds a test that we actually adopt memory from C++ objects and fixes the broken adoption from `numpy.array`

We should merge first the respective benchmarks in rootbench. The PR is ongoing here: https://github.com/root-project/rootbench/pull/195

### CODE REVIEW

File: bindings/pyroot/pythonizations/python/ROOT/pythonization/_rdataframe.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def RDataFrameAsNumpy(df, columns=None, exclude=None):
     for column in columns:
         cpp_reference = result_ptrs[column].GetValue()
         if hasattr(cpp_reference, "__array_interface__"):
-            tmp = numpy.array(cpp_reference) # This adopts the memory of the C++ object.
+            tmp = numpy.asarray(cpp_reference) # This adopts the memory of the C++ object.
```
Comment (etejedor) [inline bindings/pyroot/pythonizations/python/ROOT/pythonization/_rdataframe.py:79]: Looks good @stwunsch ! So only `asarray` adopts now and `array` copies?

File: bindings/pyroot/pythonizations/test/rdataframe_asnumpy.py [hunk 2]
```diff
@@ -280,10 +280,34 @@ def test_pickle(self):
...
         self.assertTrue(all(arr == arr2))
 
+    def test_memory_adoption_fundamental_types(self):
+        """
+        Testing the adoption of the memory from the C++ side for fundamental types
+        """
+        df = ROOT.ROOT.RDataFrame(1).Define("x", "1.0")
+        npy = df.AsNumpy(["x"])
+        pyarr = npy["x"]
+        cpparr = pyarr.result_ptr.GetValue()
```
Comment (etejedor) [inline bindings/pyroot/pythonizations/test/rdataframe_asnumpy.py:296]: This `result_ptr` is RDataFrame's result pointer that you store?

**Review by etejedor — APPROVED** [review]

File: bindings/pyroot/pythonizations/test/rdataframe_asnumpy.py (hunk 2 above)
Comment (stwunsch) [reply bindings/pyroot/pythonizations/test/rdataframe_asnumpy.py:296]: yep, exactly!

File: bindings/pyroot/pythonizations/python/ROOT/pythonization/_rdataframe.py (hunk 1 above)
Comment (stwunsch) [reply bindings/pyroot/pythonizations/python/ROOT/pythonization/_rdataframe.py:79]: I've checked the adoption when we put this into legacy pyroot, but we haven't had a test for it. For some reason it seems that the `numpy.array` constructor does not adopt anymore. I'm pretty sure it did before! However, `np.asarray` does and we check this now.

**Pull request merged**